addons/web/static/src/core/model_selector/model_selector.js

```diff
@@ -7,6 +7,21 @@ import { _t } from "@web/core/l10n/translation";
 
 import { Component, onWillStart } from "@odoo/owl";
 
+export class ModelSelectorAutoComplete extends AutoComplete {
+    selectOption(indices, params = {}) {
```

Comment:
When we click on `Start typing...`,  AutoComplete component clears the value of input field, closes the dropdown and makes an early return. But apart from that, to clear the label and make field-matchings invisible as well, we need to allow call to `onSelect` method and close the dropdown instead of an early return.

Reply:
Ok makes sense. In that case, I'll get rid of this overridden method...